lib/resque_bus/adapter.rb

```diff
@@ -11,16 +12,59 @@ def enabled!
         QueueBus::Worker.extend(::QueueBus::Adapters::Resque::RetryHandlers)
       end
 
-      def redis(&block)
+      def redis=(server)
+        @queuebus_redis = nil
+        @queuebus_redis_url = server
+      end
+
+      # This method returns the QueueBus Redis instance
+      # If it is not configured, it will default to the same instance as Resque.
+      # If it is configured, it will me intialized, memoized, and returned.
+      def queuebus_redis
+        @queuebus_redis ||
+          if @queuebus_redis_url
+            # The configuration has specified a custom server just for QueueBus.
+            # Resque doesn't offer a mechanism to create a Redis object without
+            # assignment, though, so we invoke assignment for its side-effects.
+
+            # Save the default Resque redis connection
+            default_redis = ::Resque.redis
```

Comment:
I think the dangerous method here is the parsing the the assignment does.  I'll wrap that and ensure the default gets restored.  In the case that the parsing fails, though, `@queuebus_redis` will be in an undefined state.